src/Analysis/Engine/Impl/Analyzer/DDG.cs

```diff
@@ -297,160 +302,121 @@ internal class DDG : PythonWalker {
-        /// </summary>
-        /// <param name="module">
-        /// The imported module, as returned from TryImportModule.
-        /// </param>
-        /// <param name="attribute">
-        /// Attribute within the module to resolve, already split at
-        /// dots, as returned from TryImportModule.
-        /// </param>
-        /// <param name="importAs">
-        /// The name to save the imported module or member under.
-        /// </param>
-        /// <param name="node">
-        /// If <paramref name="addRef"/> is true, this node will be
-        /// added as a reference of the imported value.
-        /// </param>
-        /// <param name="addRef">
-        /// True to add <paramref name="node"/> as a reference of the
-        /// imported value.
-        /// </param>
-        private void FinishImportModuleOrMember(ModuleReference module, IReadOnlyList<string> attribute, string name, bool addRef, Node node, NameExpression nameReference) {
-            if (AssignImportedModuleOrMember(
-                name,
-                GetImportedModuleOrMember(module, attribute, addRef, node, nameReference, attribute?.Count == 1 ? name : null),
-                addRef,
-                node,
-                nameReference
-            )) {
-                // Imports into our global scope need to enqueue modules that have imported us
-                if (Scope == GlobalScope.Scope) {
-                    GlobalScope.ModuleDefinition.EnqueueDependents();
-                }
+        private void SetVariableForImportedMember(IModule module, NameExpression importNameExpression, string importedName, NameExpression variableExpression, string variableName, bool addRef) {
+            var importedMember = module.GetModuleMember(importNameExpression, _unit, importedName, addRef, Scope, variableName);
+            if (variableExpression != null) {
+                module.GetModuleMember(variableExpression, _unit, importedName, addRef, null, null);
             }
+
+            if (importedMember.IsNullOrEmpty()) {
+                importedMember = null;
+            }
+
+            AssignImportedModuleOrMember(variableName, importedMember, addRef, importNameExpression, variableExpression);
         }
 
         public override bool Walk(FromImportStatement node) {
-            var modName = node.Root.MakeString();
-
-            if (!TryImportModule(modName, node.ForceAbsolute, out var modRef, out var bits)) {
-                _unit.DeclaringModule.AddUnresolvedModule(modName, node.ForceAbsolute);
-                return false;
+            var rootNames = node.Root.Names.Select(n => n.Name);
+            var availableImports = node.Root is RelativeModuleName relativeName
+                ? _pathResolver.GetAvailableImportsFromRelativePath(_filePath, relativeName.DotCount, rootNames)
+                : _pathResolver.GetAvailableImportsFromAbsolutePath(_filePath, rootNames);
+
+            switch (availableImports) {
```

Comment:
Does it resolve to user files, i.e. `import A` looks for A.py in the same location?

Reply:
This logic applies only to python 2. In python 3, `import a` always means that `a` is in the root folder.

Reply:
I meant, in general, for #163 which currently does not work since legacy import resolution works on libraries while user files have to be loaded when the workspace opens.

Reply:
It doesn't rely on the file being open, yes. File will be analyzed by `AstAnalysisWalker`. There may be problems though when file is open.